modules/rostests/apitests/user32/CharFuncs.c

```diff
@@ -0,0 +1,597 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for Char* functions
+ * COPYRIGHT:   Copyright 2022 Stanislav Motylkov <x86corez@gmail.com>
+ */
+
+#include "precomp.h"
+
+#include <strsafe.h>
+#include <windows.h>
+#include <ntndk.h>
+#include <pseh/pseh2.h>
+
+#define TEST_ON_VISTA  (NTDDI_VERSION >= NTDDI_VISTA)
```

Comment:
This is compile-time. Maybe it could be of interest to have this at runtime? (there may be some apitests or winetests that do a similar thing.)

Reply:
Yeah it would be better to check at runtime. I need to find the best way to query system version. :smile:

Reply:
Made the tests to decide XP vs. Vista at runtime. :+1: